docs/resilience-pipelines.md

```diff
@@ -0,0 +1,140 @@
+
+<!-- snippet: resilience-pipeline-di-usage -->
+```cs
+public static void ConfigureMyPipelines(IServiceCollection services)
+{
+    services.AddResiliencePipeline("pipeline-A", builder => builder.AddConcurrencyLimiter(100));
+    services.AddResiliencePipeline("pipeline-B", builder => builder.AddRetry(new()));
+
+    // Later, resolve the pipeline by name using ResiliencePipelineProvider<string> or ResiliencePipelineRegistry<string>
+    var pipelineProvider = services.BuildServiceProvider().GetRequiredService<ResiliencePipelineProvider<string>>();
+    pipelineProvider.GetPipeline("pipeline-A").Execute(() => { });
+}
+```
+<!-- endSnippet -->
+
+> [!NOTE]
+> In certain scenarios, such as testing, you may want to use the ResiliencePipeline.Empty instance. This instance executes the callback without applying any additional logic.
+
+## Empty resilience pipeline
+
+<!-- Explanation of the Empty Resilience Pipeline -->
+The empty resilience pipeline is a special construct that lacks any resilience strategies. You can access it through the following ways:
+
+- `ResiliencePipeline.Empty`
+- `ResiliencePipeline<T>.Empty`
+
+This is particularly useful in test scenarios where implementing resilience strategies could slow down the test execution.
+
+## Retrieving execution results with `Outcome<T>`
+
+The `ResiliencePipeline` class provides the `ExecuteOutcomeAsync` method, which is designed to never throw exceptions. Instead, it stores either the result or the exception within an `Outcome<T>` struct.
+
+<!-- snippet: resilience-pipeline-outcome -->
+```cs
+// Acquire a ResilienceContext from the pool
+ResilienceContext context = ResilienceContextPool.Shared.Get();
+
+// Execute the pipeline and store the result in an Outcome<bool>
+Outcome<bool> outcome = await pipeline.ExecuteOutcomeAsync(
+    static async (context, state) =>
+    {
+        Console.WriteLine("State: {0}", state);
+
+        try
+        {
+            await MyMethodAsync(context.CancellationToken);
+
+            // Use static utility methods from Outcome to easily create an Outcome<T> instance
+            return Outcome.FromResult(true);
+        }
+        catch (Exception e)
+        {
+            // Create an Outcome<T> instance that holds the exception
+            return Outcome.FromException<bool>(e);
+        }
+    },
+    context,
+    "my-state");
+
+// Return the acquired ResilienceContext to the pool
+ResilienceContextPool.Shared.Return(context);
+
+// Evaluate the outcome
+if (outcome.Exception is not null)
+{
+    Console.WriteLine("Execution Failed: {0}", outcome.Exception.Message);
+}
+else
+{
+    Console.WriteLine("Execution Result: {0}", outcome.Result);
+}
+```
+<!-- endSnippet -->
+
+Utilize `ExecuteOutcomeAsync` in high-performance scenarios where you wish to avoid re-throwing exceptions. Keep in mind that Polly's resilience strategies also make use of the `Outcome` struct to prevent unnecessary exception throwing.
```

Comment:
```suggestion
Use `ExecuteOutcomeAsync(...)` in high-performance scenarios where you wish to avoid re-throwing exceptions. Keep in mind that Polly's resilience strategies also make use of the `Outcome` struct to prevent unnecessary exception throwing.
```